curation-migration-datasync/eventConsumer.ts

```diff
@@ -55,3 +56,57 @@ export async function removeScheduledItem(
   // Remove association from DynamoDB
   await curatedItemModel.deleteByCuratedRecId(curatedRecord.curatedRecId);
 }
+
+/**
+ * fetches all curatedRecId related to the approvedItem's externalId.
+ * and updates them.
+ * @param eventBody
+ */
+export async function updatedApprovedItem(
+  eventBody: ApprovedItemPayload,
+  db: Knex
+) {
+  // dynamoDb will have a record of curatedRecId mapped to approvedItem
+  // only if it was previously scheduled.
+  const curatedItemModel = new CuratedItemRecordModel();
+  const curatedItems = await curatedItemModel.getByApprovedItemExternalId(
+    eventBody.approvedItemExternalId
+  );
+
+  //if dynamo returns 0 curatedItem, then the approvedItem was not scheduled before,
+  //so we can safely ignore this event.
+  if (curatedItems.length == 0) {
+    return;
+  }
+
+  //if there are approvedItem in the dynamoDb, they must be scheduled before.
+  //all scheduled item has a record in our legacy database.
+  // so we update them as per the new eventBody.
+  for (let curatedItem of curatedItems) {
+    try {
+      const dbService = new DataService(db);
+
+      if (eventBody.publisher) {
```

Comment:
is the publisher really tied with domainId? 

usually the prospect-api sends the publisher to curatedCorpusApi, which gets it from parser. so this should reamin unchanged. 

todo: look if publisher can be manually changed by curator

Reply:
we will ignore if the publisher information is updated. 

updated the code to reflect this.